### PR TITLE
EPMDJ-9248: Add render props for headers

### PR DESCRIPTION
**Pull request by dmitrii-kliushentsev**

### CODE REVIEW

File: src/components/table/table.tsx [hunk 1]
```diff
@@ -50,6 +50,7 @@ export interface Props {
   defaultFilters?: { id: string; value: string }[];
   isDefaultExpanded?: (original: any) => boolean;
   name?: string;
```
Comment (RomanAchilov) [inline src/components/table/table.tsx:52]: ?

File: src/components/table/table.tsx [hunk 2]
```diff
@@ -50,6 +50,7 @@ export interface Props {
...
+  header?: (title: string, pagesLength: number, dataLength: number) => JSX.Element;
   resultName?: string;
```
Comment (RomanAchilov) [inline src/components/table/table.tsx:54]: ?

File: src/components/table/table.tsx [hunk 3]
```diff
@@ -50,6 +50,7 @@ export interface Props {
   defaultFilters?: { id: string; value: string }[];
   isDefaultExpanded?: (original: any) => boolean;
   name?: string;
+  header?: (title: string, pagesLength: number, dataLength: number) => JSX.Element;
```
Comment (RomanAchilov) [inline src/components/table/table.tsx:53]: renderHeader

File: src/components/table/table.tsx [hunk 4]
```diff
@@ -161,7 +163,7 @@ export const Table = withErrorBoundary(({
   return (
     <>
       <div ref={ref} />
-      <TableHeader name={name} displayedResult={`Displaying ${page.length} of ${data.length} ${resultName}`} tw="mb-3" />
+      {header(name, page.length, data.length)}
```
Comment (RomanAchilov) [inline src/components/table/table.tsx:166]:
```suggestion
      {renderHeader(page.length, data.length)}
```

**Review by RomanAchilov — CHANGES_REQUESTED** [review]

File: src/components/table/table.tsx [hunk 5]
```diff
@@ -161,7 +159,7 @@ export const Table = withErrorBoundary(({
   return (
     <>
       <div ref={ref} />
-      <TableHeader name={name} displayedResult={`Displaying ${page.length} of ${data.length} ${resultName}`} tw="mb-3" />
+      {renderHeader && renderHeader(page.length, data.length)}
```
Comment (IliaSolovev) [inline src/components/table/table.tsx:162]: It must be one argument because it is a component

**Review by IliaSolovev — APPROVED** [review]

**Pull request merged**